frontend/src/pages/modelServing/customServingRuntimes/utils.ts

```diff
@@ -127,9 +127,19 @@ export const getDisplayNameFromServingRuntimeTemplate = (resource: ServingRuntim
 };
 
 export const getServingRuntimeVersionFromTemplate = (
-  resource: ServingRuntimeKind,
-): string | undefined =>
-  resource.metadata.annotations?.['opendatahub.io/runtime-version'] || undefined;
+  resource: ServingRuntimeKind | TemplateKind,
+): string | undefined => {
+  if (isTemplateKind(resource)) {
+    return (
+      resource.objects[0].metadata.annotations?.['opendatahub.io/runtime-version'] || undefined
+    );
+  }
+  return resource.metadata.annotations?.['opendatahub.io/runtime-version'] || undefined;
+};
+
+export const isTemplateKind = (
+  resource: ServingRuntimeKind | TemplateKind,
+): resource is TemplateKind => resource.kind === 'Template';
```

Comment:
```suggestion
  resource: ServingRuntimeKind | TemplateKind,
): string | undefined => {
    const servingRuntime = isServingRuntimeKind(resource)
      ? resource
      : getServingRuntimeFromTemplate(resource);
  return servingRuntime?.metadata.annotations?.['opendatahub.io/runtime-version'] || undefined;
}
```
This is what I meant. Just thought it would be easy to remove the need for the second `isTemplateKind` util

Reply:
Above I had the logic backwards so that's why it wasn't working sorry!

Reply:
oh right throws an error instead of returning undefined. Technically that is not written well. Let's stick with what you have.